### PR TITLE
New IIO C-API to read configuration attributes V2

### PR DESCRIPTION
**Pull request by guchaojie**
including 3 C-API to read scale, offset and sampling_frequency attribute

Signed-off-by: guchaojie <chao.jie.gu@intel.com>

### CODE REVIEW

**Comment by guchaojie** [comment]
@ceolin @edersondisouza @fulong82 @laykuanloon This PR is for issue #2117. All thie C-API is to get configuration attribute from device level.

File: src/lib/io/sol-iio.c [hunk 1]
```diff
@@ -1065,68 +1067,94 @@ read_channel_scan_info(struct sol_iio_channel *channel)
...
-        channel->name, device->device_id);
-    channel->scale = 1.0;
+        prefix_name, device->device_id);
+    *scale = 1.0;
 
 #undef GET_SCALE
 
 end:
-    free(channel_pure_name);
+    free(pure_name);
```
Comment (laykuanloon) [inline src/lib/io/sol-iio.c]: if L1090 GET_SCALE goto end which pure_name is NULL. This line will crash? Should add checking for not null?

**Comment by guchaojie** [comment]
@ceolin Do you have any comments on new PR?  Thanks.

File: src/lib/io/sol-iio.c [hunk 2]
```diff
@@ -1065,68 +1067,98 @@ read_channel_scan_info(struct sol_iio_channel *channel)
...
     char path[PATH_MAX];
-    char *channel_pure_name = NULL;
-    struct sol_iio_device *device = channel->device;
-    bool r;
+    char *pure_name = NULL;
+    int r;
+
+    SOL_NULL_CHECK(device, -EINVAL);
+    SOL_NULL_CHECK(prefix_name, -EINVAL);
+    SOL_NULL_CHECK(offset, -EINVAL);
```
Comment (ceolin) [inline src/lib/io/sol-iio.c]: ditto

File: src/modules/flow/iio/iio.json [hunk 3]
```diff
@@ -410,6 +495,21 @@
...
+        },
+        {
+         "data_type": "float",
+         "description": "Scale to applied to device raw readings",
+         "name": "SCALE"
+        },
+        {
+         "data_type": "float",
+         "description": "Offset to be added to device raw readings",
+        "name": "OFFSET"
```
Comment (ceolin) [inline src/modules/flow/iio/iio.json]: fix identation

**Comment by ceolin** [comment]
patch is ok, please rebase with upstream and fix these small issues that will be ready to be merged.

File: src/lib/io/sol-iio.c [hunk 4]
```diff
@@ -1065,68 +1067,98 @@ read_channel_scan_info(struct sol_iio_channel *channel)
...
     char path[PATH_MAX];
-    char *channel_pure_name = NULL;
-    struct sol_iio_device *device = channel->device;
-    bool r;
+    char *pure_name = NULL;
+    int r;
+
+    SOL_NULL_CHECK(device, -EINVAL);
+    SOL_NULL_CHECK(prefix_name, -EINVAL);
+    SOL_NULL_CHECK(scale, -EINVAL);
```
Comment (guchaojie) [inline src/lib/io/sol-iio.c]: done

File: src/modules/flow/iio/iio.json (hunk 3 above)
Comment (guchaojie) [inline src/modules/flow/iio/iio.json]: fixed

**Pull request closed without merging**